tracer/src/Datadog.Trace/Tagging/CosmosDbTags.cs

```diff
@@ -28,5 +30,49 @@ internal partial class CosmosDbTags : InstrumentationTags
 
         [Tag(Trace.Tags.OutHost)]
         public string Host { get; set; }
+
+        [Tag(Trace.Tags.OutPort)]
+        public string Port { get; set; }
+
+        public virtual void SetEndpoint(Uri endpoint)
+        {
+            Host = endpoint?.ToString();
+        }
+    }
+
+    internal partial class CosmosDbV1Tags : CosmosDbTags
+    {
+        private string _peerServiceOverride = null;
+
+        // Use a private setter for setting the "peer.service" tag so we avoid
+        // accidentally setting the value ourselves and instead calculate the
+        // value from predefined precursor attributes.
+        // However, this can still be set from ITags.SetTag so the user can
+        // customize the value if they wish.
+        [Tag(Trace.Tags.PeerService)]
+        public string PeerService
+        {
+            get => _peerServiceOverride ?? DatabaseId ?? Host;
+            private set => _peerServiceOverride = value;
+        }
+
+        [Tag(Trace.Tags.PeerServiceSource)]
+        public string PeerServiceSource
+        {
+            get
+            {
+                return _peerServiceOverride is not null
+                        ? "peer.service"
+                        : DatabaseId is not null
+                            ? "db.instance"
+                            : "network.destination.name";
```

Comment:
```suggestion
                            : "out.host";
```

Reply:
Thanks! Fixed in de94b98e79c09779da6ad32a8f3923286149a73f

Reply:
Actually, had to repush since my branch still used `db.instance` as a source. That's updated to `db.name` in 8fe65bf4be32b057530a30d614a94ba16d56485c